presentation/feed/src/main/java/com/hilingual/presentation/feed/FeedViewModel.kt

```diff
@@ -39,31 +40,53 @@ import javax.inject.Inject
 @HiltViewModel
 internal class FeedViewModel @Inject constructor(
     private val feedRepository: FeedRepository,
-    private val diaryRepository: DiaryRepository
+    private val diaryRepository: DiaryRepository,
+    private val userRepository: UserRepository
 ) : ViewModel() {
     private val _uiState = MutableStateFlow(FeedUiState())
     val uiState: StateFlow<FeedUiState> = _uiState.asStateFlow()
 
     private val _sideEffect = MutableSharedFlow<FeedSideEffect>()
     val sideEffect: SharedFlow<FeedSideEffect> = _sideEffect.asSharedFlow()
 
-    fun loadFeedData() {
+    fun loadInitialFeedData() {
+        getMyProfile()
         getRecommendFeeds()
         getFollowingFeeds()
     }
 
+    fun onFeedRefresh(tab: FeedTab) {
+        when (tab) {
+            FeedTab.RECOMMEND -> getRecommendFeeds()
+            FeedTab.FOLLOWING -> getFollowingFeeds()
+        }
+    }
+
     fun readAllFeed() {
         viewModelScope.launch {
             emitToastSideEffect("피드의 일기를 모두 확인했어요.")
         }
     }
 
+    private fun getMyProfile() {
+        viewModelScope.launch {
+            userRepository.getUserLoginInfo().onSuccess { myInfo ->
+                _uiState.update {
+                    it.copy(
+                        myProfileUrl = myInfo.profileImg
+                    )
+                }
+            }.onLogFailure { }
```

Comment:
```suggestion
            userRepository.getUserLoginInfo()
                .onSuccess { myInfo ->
                    _uiState.update {
                        it.copy(
                            myProfileUrl = myInfo.profileImg
                        )
                    }
                }
                .onLogFailure { }
```
암묵적인 룰이랄까요..?? 코드가 엄청 짧으면 한 줄로 하는게 더 가독성이 좋지만 지금처럼 람다 포함되어 있거나 코드가 조금이라도 길어지면 줄바꿈을 하는게 가독성에 더 좋습니다! (공식적으로도 이 방식을 더 추천한다고 하네요!)